x-pack/plugins/ml/public/components/annotations_table/annotations_table.js

```diff
@@ -0,0 +1,342 @@
+        return (
+          <EuiCallOut
+            title={this.state.errorMessage}
+            color="danger"
+            iconType="cross"
+          />
+        );
+      }
+    }
+
+    const annotations = this.props.annotations || this.state.annotations;
+
+    if (annotations.length === 0) {
+      return (
+        <EuiCallOut
+          title="No annotations created for this job"
+          iconType="iInCircle"
+        >
+          <p>
+            To create an annotation,
+            open the <EuiLink onClick={() => this.openSingleMetricView()}>Single Metric Viewer</EuiLink>
+          </p>
+        </EuiCallOut>
+      );
+    }
+
+    function renderDate(date) { return formatDate(date, TIME_FORMAT); }
+
+    const columns = [
+      {
+        field: 'annotation',
+        name: 'Annotation',
+        sortable: true
+      },
+      {
+        field: 'timestamp',
+        name: 'From',
+        dataType: 'date',
+        render: renderDate,
+        sortable: true,
+        width: '180px'
+      },
+      {
+        field: 'end_timestamp',
+        name: 'To',
+        dataType: 'date',
+        render: renderDate,
+        sortable: true,
+        width: '180px'
+      },
+      {
+        field: 'create_time',
+        name: 'Creation date',
+        dataType: 'date',
+        render: renderDate,
+        sortable: true,
+        width: '180px'
+      },
+      {
+        field: 'create_username',
+        name: 'Created by',
+        sortable: true,
+        width: '180px'
+      },
+      {
+        field: 'modified_time',
+        name: 'Last modified date',
+        dataType: 'date',
+        render: renderDate,
+        sortable: true,
+        width: '180px'
+      },
+      {
+        field: 'modified_username',
+        name: 'Last modified by',
+        sortable: true,
+        width: '180px'
+      },
+    ];
+
+    if (isNumberBadgeVisible) {
+      columns.unshift({
+        field: 'key',
+        name: '#',
+        width: '50px',
+        render: (key) => {
+          return (
+            <EuiBadge color="default">
+              {key}
+            </EuiBadge>
+          );
+        }
+      });
+    }
+
+    if (isSingleMetricViewerLinkVisible) {
+      const openInSingleMetricViewerText = 'Open in Single Metric Viewer';
+      columns.push({
+        align: RIGHT_ALIGNMENT,
+        width: '60px',
+        name: 'View',
+        render: (annotation) => (
+          <EuiToolTip
+            position="bottom"
+            content={openInSingleMetricViewerText}
+          >
+            <EuiButtonIcon
+              onClick={() => this.openSingleMetricView(annotation)}
+              iconType="stats"
+              aria-label={openInSingleMetricViewerText}
+            />
+          </EuiToolTip>
+        )
+      });
+    }
+
+    const getRowProps = (item) => {
+      return {
+        onMouseOver: () => this.onMouseOverRow(item),
+        onMouseLeave: () => this.onMouseLeaveRow()
```

Comment:
Same here - shouldn't need to wrap with an arrow function.

Reply:
Done in e251767910.